### ISSUE TITLE
template error while templating string

### ISSUE THREAD

**Issue by smaboshe** [louim/bedrock-site-protect#19]
Hello, @louim,

Hope you are well. I have been running into this error now and then when provisioning Trellis sites that use `bedrock-site-protect`.

```
TASK [bedrock-site-protect : Set htpasswd] **************************************************************************************************************************************************************************************
System info:
  Ansible 2.7.5; Linux
  Trellis 1.0.0: December 27th, 2018
---------------------------------------------------
template error while templating string: unexpected char u'$' at 52. String:
^8oea`nE!ke!rQ9-6,g]bQ:g<Q#9@7778Ayrn/$;N{{BwrOt*rtz$?a#tvFE|Hu}
fatal: [example.com]: FAILED! => {}
```

The offending string will be from `vault.yml` generated by https://roots.io/salts.html

Generating a new set of salts usually corrects the issue. Is there a way to clear this up permanently?

**Comment by alex-galey** [comment]
Same issue here bedrock-site-protect doesn't like salt

It seems to happen on when one salt contains # symbol 2 times, I removed all # from the salts whether I am not sure if the part of the salt after the # will be considered as a comment.

This custom role was also there during provisioning and didn't complain about these strings [Cloudflare Origin CA tasks](https://github.com/TypistTech/trellis-cloudflare-origin-ca/blob/master/tasks/certificates.yml)

@louim : can you give us your insight here ?

**Comment by alex-galey** [comment]
Perhaps it's an old ansible issue : https://github.com/ansible/ansible/issues/16443

It is documented in Ansible docs here https://docs.ansible.com/ansible/latest/user_guide/playbooks_advanced_syntax.html : 

##YAML 
`my_unsafe_variable: !unsafe 'this variable has {{ characters that should not be treated as a jinja2 template'`

## Template j2
```
{% raw %}
...
{% endraw %}
```

**Comment by louim** [comment]
Hey!

Interesting issue, seem's like in @smaboshe version of the salt, it's caused by the `{{` making Ansible think there's a variable there.

In your salt, did you have `{` before the `#`?

I'm not exactly sure where in the whole process Ansible is trying to template the strings, I'll need to dig deeper.

**Comment by gblakeman** [comment]
I just ran into this myself with one of my environments.

The issue was initially created by a `{{` in one of my salts. I changed it, and then a different salt further down the chain tripped the issue with `{%`. Changing that one finally solved the issue. 🤷🏻‍♂️

**Comment by louim** [comment]
Hey, thanks for the additional report. I had a look today and it’s a complicated issue due to the fact that we’re doing a two layer deep loop. I tried a couple of things, but none of them worked 😢. 

I don’t have an easy fix, but I’m still exploring different ways to fix the problem. There’s hope 🤞

**Comment by gblakeman** [comment]
@louim Thanks for looking into it! At least it’s an easy, temporary fix now that we know what to look out for in the generated salts. 🙌

**Comment by louim** [comment]
This issue should resolve itself (no need to update` bedrock-site-protect`) once https://github.com/roots/trellis/pull/1071 is merged into Trellis. 🎉

**Comment by dalepgrant** [comment]
Just to add a bit more to this, I can confirm that having `{#`, `{{` or `{%` in the salts causes the same error. Haven't come across any others, we've got a note in our `vault.yml` reminding us to check for those three when adding new salts.

**Comment by louim** [comment]
Hey all! 

The PR containing the fix has just been merged in Trellis master. This issue should now be fixed once you use the latest Trellis. 

Thanks for the report and all the help investigating! 🎉
